editor/components/inserter/menu.js

```diff
@@ -29,32 +78,170 @@ class InserterMenu extends wp.element.Component {
 		return () => {
 			this.props.onInsertBlock( slug );
 			this.props.onSelect();
-			this.setState( { filterValue: '' } );
+			this.setState( {
+				filterValue: '',
+				currentFocus: null
+			} );
 		};
 	}
 
-	render() {
-		const { position = 'top' } = this.props;
-		const blocks = wp.blocks.getBlocks();
-		const isShownBlock = block => block.title.toLowerCase().indexOf( this.state.filterValue.toLowerCase() ) !== -1;
-		const blocksByCategory = blocks.reduce( ( groups, block ) => {
-			if ( ! isShownBlock( block ) ) {
-				return groups;
-			}
-			if ( ! groups[ block.category ] ) {
-				groups[ block.category ] = [];
+	getVisibleBlocks( blockTypes ) {
+		return blockTypes.filter( this.isShownBlock );
+	}
+
+	getCategoryIndex( item ) {
```

Comment:
Granted this is personal preference. I like minimising the number of class methods to the necessary, for example this is only used in `sortBlocksByCategory`, I could write it as an inline arrow function.

`const getCategoryIndex = ( item ) => this.categories.findIndex( ( category ) => category.slug === item.slug );`